wheatley/tower.py

```diff
@@ -245,8 +245,18 @@ def _on_size_change(self, data):
         """ Callback called when the number of bells in the room changes. """
         new_size = data["size"]
         if new_size != self.number_of_bells:
-            self._assigned_users = {}
+            # Remove the user who's bells have been removed (so that returning to a stage doesn't make
+            # Wheatley think the bells are still assigned)
+            bells_to_unassign = []
+            for bell in self._assigned_users:
+                if bell.number > new_size:
+                    bells_to_unassign.append(bell)
```

Comment:
To reduce the amount of indentation
```suggestion
            bells_to_unassign = [bell for bell in self._assigned_users if bell.number > new_size]
```

Reply:
Or perhaps the whole thing in one line would work?
```python
self._assigned_users = {bell: user for (bell, user) in self._assigned_users.items() if bell.number <= new_size }
```

Reply:
Done